### ISSUE TITLE
Error: Please provide an index for your component [no-index]

### ISSUE THREAD

**Issue by mervynteo** [matteodem/meteor-easy-search#317]
Heres the full error 

#### (Error message)
```javascript
//Uncaught Error: Match error: Failed Match.OneOf or Match.Optional validation
Error: Please provide an index for your component [no-index]
    at EachComponent.onCreated (base.js:37) // if (_.isEmpty(indexes))
    at EachComponent.onCreated (single-index.js:11)
    at registerHooks.onCreated (peerlibrary_blaze-components.js:650)
    at blaze.js:3302
    at Function.Template._withTemplateInstanceFunc (blaze.js:3643)
    at fireCallbacks (blaze.js:3298)
    at null.<anonymous> (blaze.js:3378)
    at fireCallbacks (blaze.js:1924)
    at Object.Tracker.nonreactive (tracker.js:589)
    at blaze.js:1921
```
### JS client
```javascript
// on both client and server 
postsIndex    = new EasySearch.Index({
  collection  : Posts,
  fields      : ['url', 'message', 'title', 'author'], 
  engine      : new EasySearch.MongoDB()
});
commentsIndex = new EasySearch.Index({
  collection  : Comments,
  fields      : ['author', 'body'], 
  engine      : new EasySearch.MongoDB()
});

//on the client 
Template.searchPosts.onCreated( function() {

  Tracker.autorun(function () {              
    cursor = postsIndex.search();         // this causes the Failed Match.OneOf or Match.Optional validation error 
    cursor = commentsIndex.search();    // this causes the Failed Match.OneOf or Match.Optional validation error 

    console.log(cursor.fetch()); 
    console.log(cursor.count()); 

    Meteor.subscribe('searchPosts', function(){
      console.log("search subscription is ready");
      //console.log(Posts.find().count());
    });

  });
});

Template.searchPosts.helpers({
  index: function () {
    return [postsIndex, commentsIndex];   // corrected from PostsIndex and CommentsIndex
  },
  posts: function () {
    return Posts.find();
  }
});

//on server - this publish causing errors (like overlapping publish or something
/*
Meteor.publish('searchPosts', function(){
  return Posts.find();
 })

Meteor.publish('searchPosts', function(searchString){
  check(searchString, String);
  return index.search(searchString).mongoCursor;
})
*/
```
### HTML
```html
{{> EasySearch.Input indexes=index}}    <!-- changed from 'index=index' -->

     {{#EasySearch.Each index=postsIndex}}
         {{> postItem}}
     {{/EasySearch.Each}}

      {{#EasySearch.Each index=commentsIndex}}
         {{> postItem}}
      {{/EasySearch.Each}}


<!--  previous version
{{#EasySearch.Each index=index}}        
     {{> postItem}}
{{/EasySearch.Each}}
-->
```

**Comment by matteodem** [comment]
You need to use `indexes=index` for several indexes. Also be aware that you cannot use multiple `indexes` on EasySearch.Each

**Comment by mervynteo** [comment]
Thanks for the prompt reply! Have edited and got the same error. Appreciate your help to enlighten me on which parts are bad?

**Comment by mervynteo** [comment]
Thanks for the prompt reply! Have edited as above and got the same error. Appreciate if you can help enlighten me on which parts are bad? It would be best if theres a sample within the leaderboard demo.

**Comment by matteodem** [comment]
In your code example `PostIndex` is being used in the template and defined it's as `postIndex` (small p).

**Comment by mervynteo** [comment]
Corrected it, the error remains unfortunately. The error is saying I dont have an index for the component and something broke on blaze. Could it be an issue with my publish and subscribe? like it should be lines i commented in the publish lines above?

**Comment by matteodem** [comment]
I'll try it out with the leaderboard and see if it throws the same error.

Please create a reproduction repository in case that I cannot reproduce it.

**Comment by mervynteo** [comment]
Alright sure, thanks a lot!
Ill try to come up with a repo too as the Comments are not nested with the Posts but rather as a seperate collection.

**Comment by matteodem** [comment]
I couldn't reproduce it with the leaderboard example.

**Comment by mervynteo** [comment]
Hi Matt, I managed to repro the matching error. You will get the main error of no-index when u click url link to new page, then right click for browser devtools (inspect element). 
To get the matching error, just comment out the testing codes including the console.log. 

http://meteorpad.com/pad/bH8eiH9Tjv3gk48po/Dual-index%20search

**Comment by mervynteo** [comment]
@matteodem any luck with it? Was wondering if its an issue with the schema. I.e if comments were directly under value pair of post.

**Comment by matteodem** [comment]
Wasn't able to get a look at it, will do over the weekend though

**Comment by matteodem** [comment]
The problem is that `commentsIndex` and `postsIndex` are not defined as helpers in your app.js, fix is:

```javascript
//Template helper 
Template.searchPosts.helpers({
  index: function () {
    return [postsIndex, commentsIndex];   
  },
  posts: function () {
    return Posts.find();
  },
  comments: function () {
    return Comments.find({postId: this._id});
  },
  // THIS IS MISSING
  postsIndex: function () { return postsIndex; },
  commentsIndex: function () { return commentsIndex; }
});

```

**Comment by mervynteo** [comment]
Thanks @matteodem theres finally search results appearing!! 
Need to check with you on 2 things: 
1) There seems to be odd search results. Like repeated past data that was supposed to be deleted with the Meteor reset. For example, my seed data has A, B and C. I did 2 resets. So now before I search, theres A, B and C...followed by A1, A2, B1, B2 and C1, C2 where those are past data I have deleted with the reset. 
2) The original items when clicked goes straight into the item detail (like a url link click). However that doesnt work with the search result. How should I tweak the html to adjust for this?
